src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/proposal/quickfix/RemoveAnnotationConflictQuickFix.java

```diff
@@ -0,0 +1,132 @@
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiModifierListOwner;
+import com.intellij.psi.PsiVariable;
+import com.intellij.psi.util.PsiTreeUtil;
+import io.openliberty.tools.intellij.lsp4jakarta.lsp4ij.codeAction.proposal.DeleteAnnotationProposal;
+import io.openliberty.tools.intellij.lsp4mp4ij.psi.core.java.codeaction.JavaCodeActionContext;
+import io.openliberty.tools.intellij.lsp4mp4ij.psi.core.java.corrections.proposal.ChangeCorrectionProposal;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * QuickFix for removing annotations. Modified from
+ * https://github.com/eclipse/lsp4mp/blob/6f2d700a88a3262e39cc2ba04beedb429e162246/microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/java/codeaction/InsertAnnotationMissingQuickFix.java
+ *
+ * @author Angelo ZERR
+ *
+ */
+public class RemoveAnnotationConflictQuickFix {
+    private final String[] annotations;
+
+    protected final boolean generateOnlyOneCodeAction;
+
+    /**
+     * Constructor for remove annotation quick fix.
+     *
+     * <p>
+     * The participant will generate a CodeAction per annotation.
+     * </p>
+     *
+     * @param annotations list of annotation to insert.
+     */
+    public RemoveAnnotationConflictQuickFix(String... annotations) {
+        this(false, annotations);
+    }
+
+    /**
+     * Constructor for remove annotation quick fix.
+     *
+     * @param generateOnlyOneCodeAction true if the participant must generate a
+     *                                  CodeAction which insert the list of
+     *                                  annotation and false otherwise.
+     * @param annotations               list of annotation to insert.
+     */
+    public RemoveAnnotationConflictQuickFix(boolean generateOnlyOneCodeAction, String... annotations) {
+        this.generateOnlyOneCodeAction = generateOnlyOneCodeAction;
+        this.annotations = annotations;
+    }
+
+    public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic) {
+        PsiElement node = context.getCoveredNode();
+        PsiElement parentType = getBinding(node);
+        if (parentType != null) {
+            List<CodeAction> codeActions = new ArrayList<>();
+            removeAnnotations(diagnostic, context, parentType, codeActions);
+            return codeActions;
+        }
+        return null;
+
+    }
+
+    protected void removeAnnotations(Diagnostic diagnostic, JavaCodeActionContext context, PsiElement parentType,
+                                     List<CodeAction> codeActions) {
+        if (generateOnlyOneCodeAction) {
+            removeAnnotation(diagnostic, context, parentType, codeActions, annotations);
+        } else {
+            for (String annotation : annotations) {
+                removeAnnotation(diagnostic, context, parentType, codeActions, annotation);
```

Comment:
Before anybody mentions I just noticed that in this loop we need to create a new context so that will come in a future pull request.

Reply:
Never mind, I can add it to this pull request.